extensions/resource-deployment/src/ui/deploymentInputDialog.ts

```diff
@@ -63,11 +73,46 @@ export class DeploymentInputDialog extends DialogBase {
 		setModelValues(this.inputComponents, model);
 		if (instanceOfNotebookBasedDialogInfo(this.dialogInfo)) {
 			model.setEnvironmentVariables();
-			this.notebookService.launchNotebook(this.dialogInfo.notebook).then(() => { }, (error) => {
-				vscode.window.showErrorMessage(error);
-			});
+			if (this.dialogInfo.runNotebook) {
+				this.executeNotebook(this.dialogInfo);
+			} else {
+				this.notebookService.launchNotebook(this.dialogInfo.notebook).then(() => { }, (error) => {
+					vscode.window.showErrorMessage(error);
+				});
+			}
 		} else {
 			vscode.commands.executeCommand(this.dialogInfo.command, model);
 		}
 	}
+
+	private executeNotebook(notebookDialogInfo: NotebookBasedDialogInfo): void {
+		azdata.tasks.startBackgroundOperation({
+			displayName: notebookDialogInfo.taskName!,
+			description: notebookDialogInfo.taskName!,
+			isCancelable: false,
+			operation: async op => {
+				op.updateStatus(azdata.TaskStatus.InProgress);
+				const notebook = await this.notebookService.getNotebook(notebookDialogInfo.notebook);
+				const result = await this.notebookService.executeNotebook(notebook);
+				if (result.succeeded) {
+					op.updateStatus(azdata.TaskStatus.Succeeded);
```

Comment:
log output strings to deployment output channel, same with error codes/strings/messages.